core/src/sumeragi/mod.rs

```diff
@@ -188,42 +178,50 @@ impl Sumeragi {
         block_hash: HashOf<VersionedCommittedBlock>,
     ) -> Vec<VersionedCommittedBlock> {
         self.internal
-            .wsv
+            .wsv_for_public_use
             .lock()
             .unwrap()
             .blocks_after_hash(block_hash)
     }
 
     pub fn get_random_peer_for_block_sync(&self) -> Option<Peer> {
-        use rand::{prelude::SliceRandom, SeedableRng};
+        use rand::{prelude::SliceRandom, RngCore, SeedableRng};
 
         let rng = &mut rand::rngs::StdRng::from_entropy();
-        self.internal
-            .wsv
+        let mut vec = self
+            .internal
+            .current_online_peers
             .lock()
-            .unwrap()
-            .peers()
-            .choose(rng)
-            .cloned()
+            .expect("lock on online peers for get random peer")
+            .iter()
+            .map(|peer| Peer::new((*peer).clone()))
+            .collect::<Vec<Peer>>();
+        if vec.is_empty() {
+            return None;
+        }
+        vec.sort();
```

Comment:
```suggestion
        if vec.is_empty() {
            None
        } else {
             vec.sort();
             Some(...)
       }
```

You're mixing styles. If you're short circuiting, it's not immediately obvious that things outside of the `if` are alternatives to the `if`. This is obvious to a C programmer (who would consider this an anti-pattern), but everyone else would find the `else` style more explicit.

Reply:
Also consider sorting after the `iter()` call.

Reply:
sure